server/stream.go

```diff
@@ -1047,6 +1047,9 @@ func (mset *stream) update(config *StreamConfig) error {
 		}
 	}
 
+	mset.js.releaseStreamResources(&ocfg)
+	mset.js.reserveStreamResources(cfg)
```

Comment:
Have you checked the lock ordering here?  This change wants the jetstream lock while holding the mset lock.
In the create/destroy case that happens outside mset.mu.